test/unittests/execute_floating_point_test.cpp

```diff
@@ -66,4 +69,209 @@ TEST(execute_floating_point, f64_add)
+struct ConversionPairWasmTraits<double, uint32_t>
+{
+    static constexpr auto opcode_name = "i32_trunc_f64_u";
+    static constexpr auto opcode = Instr::i32_trunc_f64_u;
+    static constexpr auto src_valtype = ValType::f64;
+    static constexpr auto dst_valtype = ValType::i32;
+};
+template <>
+struct ConversionPairWasmTraits<float, int64_t>
+{
+    static constexpr auto opcode_name = "i64_trunc_f32_s";
+    static constexpr auto opcode = Instr::i64_trunc_f32_s;
+    static constexpr auto src_valtype = ValType::f32;
+    static constexpr auto dst_valtype = ValType::i64;
+};
+template <>
+struct ConversionPairWasmTraits<float, uint64_t>
+{
+    static constexpr auto opcode_name = "i64_trunc_f32_u";
+    static constexpr auto opcode = Instr::i64_trunc_f32_u;
+    static constexpr auto src_valtype = ValType::f32;
+    static constexpr auto dst_valtype = ValType::i64;
+};
+template <>
+struct ConversionPairWasmTraits<double, int64_t>
+{
+    static constexpr auto opcode_name = "i64_trunc_f64_s";
+    static constexpr auto opcode = Instr::i64_trunc_f64_s;
+    static constexpr auto src_valtype = ValType::f64;
+    static constexpr auto dst_valtype = ValType::i64;
+};
+template <>
+struct ConversionPairWasmTraits<double, uint64_t>
+{
+    static constexpr auto opcode_name = "i64_trunc_f64_u";
+    static constexpr auto opcode = Instr::i64_trunc_f64_u;
+    static constexpr auto src_valtype = ValType::f64;
+    static constexpr auto dst_valtype = ValType::i64;
+};
+
+template <typename SrcT, typename DstT>
+struct ConversionPair : ConversionPairWasmTraits<SrcT, DstT>
+{
+    using src_type = SrcT;
+    using dst_type = DstT;
+};
+
+struct ConversionName
+{
+    template <typename T>
+    static std::string GetName(int /*unused*/)
+    {
+        return T::opcode_name;
+    }
+};
+
+template <typename T>
+class execute_floating_point_trunc : public testing::Test
+{
+};
+
+using TruncPairs = testing::Types<ConversionPair<float, int32_t>, ConversionPair<float, uint32_t>,
+    ConversionPair<double, int32_t>, ConversionPair<double, uint32_t>,
+    ConversionPair<float, int64_t>, ConversionPair<float, uint64_t>,
+    ConversionPair<double, int64_t>, ConversionPair<double, uint64_t>>;
+TYPED_TEST_SUITE(execute_floating_point_trunc, TruncPairs, ConversionName);
+
+TYPED_TEST(execute_floating_point_trunc, trunc)
+{
+    using FloatT = typename TypeParam::src_type;
+    using IntT = typename TypeParam::dst_type;
+    using FloatLimits = std::numeric_limits<FloatT>;
+    using IntLimits = std::numeric_limits<IntT>;
+
+    /* wat2wasm
+    (func (param f32) (result i32)
+      local.get 0
+      i32.trunc_f32_s
+    )
+    */
+    auto wasm = from_hex("0061736d0100000001060160017d017f030201000a070105002000a80b");
+
+    // Find and replace changeable values: types and the conversion instruction.
+    auto& param_type = wasm[wasm.find(static_cast<uint8_t>(ValType::f32))];
+    auto& result_type = wasm[wasm.find(static_cast<uint8_t>(ValType::i32))];
```

Comment:
Perhaps adding assertion that there are only a single occurrence of these valtypes?